src/CarbonAware.DataSources/CarbonAware.DataSources.Json/src/Configuration/JsonDataConfiguration.cs

```diff
@@ -0,0 +1,49 @@
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace CarbonAware.DataSources.Json.Configuration;
+
+/// <summary>
+/// A configuration class for holding Json Data config values.
+/// </summary>
+public class JsonDataConfiguration
+{
+    private const string BaseDir = "data-files";
+    private const string DefaultDataFile = "test-data-azure-emissions.json";
+    private const string RegExDir = @"^[-/a-zA-Z_\d ]*$";
+    private string? dataFileLocation;
+    private string assemblyDirectory;
+
+    public const string Key = "JsonData";
```

Comment:
From the user perspective they are configuring the `JsonDataSource`, so to match their expectations as well as how they configure other components like `WattTimeClient` I would recommend we update this key and change the class name to `JsonDataSourceConfiguration`